extensions/amp-a4a/0.1/amp-a4a.js

```diff
@@ -415,13 +436,15 @@ export class AmpA4A extends AMP.BaseElement {
         // If we got as far as getting a URL, then load the ad, but note the
         // error.
         if (this.adUrl_) {
+          this.lifecycleReporter_.sendPing('renderViaIframe');
           this.renderViaIframe_(true);
         }
         throw rendered;
-      };
+      }
       if (!rendered) {
         // Was not AMP creative so wrap in cross domain iframe.  layoutCallback
         // has already executed so can do so immediately.
+        this.lifecycleReporter_.sendPing('renderViaIframe');
```

Comment:
Could be, I guess.  But I didn't put any of the other `sendPing`s in the corresponding methods.  I could, I suppose, put them all there.  But having them in the promise chain made it easier to ensure that I was hitting all the key points.  Do you feel strongly?